src/course-unit/xblock-container-iframe/index.tsx

```diff
@@ -1,57 +1,183 @@
+import { useNavigate } from 'react-router-dom';
 
-import { IFRAME_FEATURE_POLICY } from '../constants';
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import ConfigureModal from '../../generic/configure-modal/ConfigureModal';
+import { copyToClipboard } from '../../generic/data/thunks';
+import { COURSE_BLOCK_NAMES, IFRAME_FEATURE_POLICY } from '../../constants';
+import { messageTypes, COMPONENT_TYPES } from '../constants';
+import { fetchCourseUnitQuery } from '../data/thunk';
 import { useIframe } from '../context/hooks';
 import { useIFrameBehavior } from './hooks';
 import messages from './messages';
 
-/**
- * This offset is necessary to fully display the dropdown actions of the XBlock
- * in case the XBlock does not have content inside.
- */
-const IFRAME_BOTTOM_OFFSET = 220;
+import {
+  XBlockContainerIframeProps,
+  XBlockDataTypes,
+  MessagePayloadTypes,
+} from './types';
 
-interface XBlockContainerIframeProps {
-  blockId: string;
-}
-
-const XBlockContainerIframe: FC<XBlockContainerIframeProps> = ({ blockId }) => {
+const XBlockContainerIframe: FC<XBlockContainerIframeProps> = ({
+  courseId, blockId, unitXBlockActions, courseVerticalChildren, handleConfigureSubmit,
+}) => {
   const intl = useIntl();
   const iframeRef = useRef<HTMLIFrameElement>(null);
-  const { setIframeRef } = useIframe();
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const [isDeleteModalOpen, openDeleteModal, closeDeleteModal] = useToggle(false);
+  const [isConfigureModalOpen, openConfigureModal, closeConfigureModal] = useToggle(false);
+  const { setIframeRef, sendMessageToIframe } = useIframe();
+  const [currentXBlockId, setCurrentXBlockId] = useState<string | null>(null);
+  const [currentXBlockData, setCurrentXBlockData] = useState<any>({});
+  const [courseXBlockIframeOffset, setCourseXBlockIframeOffset] = useState<number>(0);
+
+  const iframeUrl = useMemo(() => `${getConfig().STUDIO_BASE_URL}/container_embed/${blockId}`, [blockId]);
+
+  useEffect(() => {
+    setIframeRef(iframeRef);
+  }, [setIframeRef]);
+
+  useEffect(() => {
+    if (currentXBlockId) {
+      const foundXBlockInfo = courseVerticalChildren?.find(xblock => xblock.blockId === currentXBlockId);
+      if (foundXBlockInfo) {
+        const { name, userPartitionInfo, blockType } = foundXBlockInfo;
+
+        setCurrentXBlockData({
+          category: COURSE_BLOCK_NAMES.component.id,
+          displayName: name,
+          userPartitionInfo,
+          showCorrectness: 'always',
+          blockType,
+          id: currentXBlockId,
+        });
+      }
+    }
+  }, [isConfigureModalOpen, currentXBlockId, courseVerticalChildren]);
+
+  const handleRefreshIframe = useCallback(() => {
+    // TODO: this artificial delay is a temporary solution
+    // to ensure the iframe content is properly refreshed.
+    setTimeout(() => {
+      sendMessageToIframe(messageTypes.refreshXBlock, null);
+    }, 1000);
+  }, [sendMessageToIframe]);
```

Comment:
[inform]: we plan to abandon such reboots after implementing the tree comparison algorithm on the backbone.js side

Reply:
What would that look like? I generally think we need to try to dramatically reduce the amount of code we have on the backbone side, not add major new functionality to it.

Reply:
We assumed there might be a comparison algorithm that compares the response containing xblocks information with the current backbone xblocks model. However, this idea turned out to be incorrect since we do not have separate models for xblocks on the backbone side. At the moment, a draft of a working solution is presented in these draft PRs.
- [edx-platform](https://github.com/openedx/edx-platform/pull/35959/commits/db5af74ce6a6ef2955950d2ee6986b7d11ab8151)
- [frontend-app-authoring](https://github.com/openedx/frontend-app-authoring/pull/1544/commits/61f957ebf653c3bc28653a76b1efe73683d28ac7)